src/api/app/models/bs_request.rb

```diff
@@ -31,11 +31,16 @@ class SaveError < APIException
   ]
 
   scope :to_accept, -> { where(state: 'new').where('accept_at < ?', DateTime.now) }
+
   # Scopes for collections
   scope :with_actions, -> { joins(:bs_request_actions).distinct.order(priority: :asc, id: :desc) }
   scope :in_states, ->(states) { where(state: states) }
   scope :with_types, ->(types) { where('bs_request_actions.type in (?)', types) }
+  scope :with_review, -> (review, project) { includes(:reviews).where("reviews.state = 'review' and reviews.by_project = ?", project).references(:reviews) }
```

Comment:
@ChrisBr 

> Why is the review variable necessary?

Because I use it here, so the code there doesn't look really ugly and complicate, it is much readable with this:

https://github.com/openSUSE/open-build-service/pull/2899/files#diff-06133ea6068a576c1f21fe39cfdf8536R1198


@bgeuken 
> I think part of this should be a scope of the review mode

But how? I need the `includes` and the `references` and also set the project. So I don't get what you are exactly proposing.